Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM node:8
+
+# install yarn
+RUN apt-get update && apt-get install -y apt-transport-https \
+ && curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+ && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+ && apt-get update && apt-get install -y yarn \
+ && rm /var/lib/apt/lists/* -fR
+
+# install ffmpeg
+RUN echo "deb http://ftp.uk.debian.org/debian jessie-backports main" >/etc/apt/sources.list.d/jessie-backports.list \
+ && apt-get update \
+ && apt-get -y install ffmpeg \
+ && rm /var/lib/apt/lists/* -fR
+
+WORKDIR /usr/src/app
+
+COPY package.json yarn.lock ./
```

Comment:
It is preferable to run Yarn after successful Docker deployment. This would keep containers light.

Reply:
Disagreeing with both comments. Downloading a binary release is very diffrent from building the whole thing and has very different properties. For instance, one might wish to build from source and have a repeatable result.

Also, running yarn after deployment keeps the image light but definitely not the container as files are still stored whenever the thing is running. Also, you then lose the interesting stateless property of the image.

I would maybe go for multiple Dockerfile, at least one building from source and one embedding the latest release.

Reply:
Agreeing with @kaiyou. I was hoping the repository owner might add an [automated build](https://docs.docker.com/docker-hub/builds/) on the Docker hub, thus automatically building a new release for every commit. We could also use Git tags in order to let the Docker hub build and tag images for every release. So it makes sense to build from source.

Concerning the yarn comment, it is desirable to have an image which can be deployed seamlessly on a Docker host / a Docker swarm cluster. In order for that to work, we need a ready-made, complete image. As mentioned by @kaiyou, in practice this does not save space at all.

That said, we could have a different Dockerfile for development and production. In my opinion, this doesn't help much though. In either case, you only need to rebuild the image when dependencies change. (The current code from your working directory can be bind-mounted in development.)

Reply:
Afaik, Docker automated builds do not support setting the Dockerfile path, but the build root path instead, which is not compatible with the current structure. Still, an automated build could be delegated to Travis or any CI platform :)

Reply:
Yes it does now, actually (if we're talking about setting the Dockerfile path in hub.docker.com). I know since I've set it up, for one project of mine ;)

Reply:
I think that setting the "Dockerfile location" in a Docker hub build will set the Docker build context, not the Dockerfile path, which prevents from adding files in parent directories and will make ``COPY . ./`` invalid.